.github/workflows/main.yml

```diff
@@ -0,0 +1,146 @@
+name: process-balancer-main
+
+
+on:
+  pull_request:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+.[0-9a-z]+
+      - v[0-9]+.[0-9]+.[0-9]+
+
+
+jobs:
+
+  tests:
+    name: Tests
+    runs-on: [self-hosted, ubuntu-t3.medium]
+    container:
+      image: public.ecr.aws/q0j1f2t0/ruby:2.5-bionic
+      options: --user 1000
+      env:
+        BUNDLE_PATH: vendor/bundle
+        BUNDLE_VERSION: 1.17.3
+        BUNDLE_JOBS: 4
+        BUNDLE_RETRY: 3
+        RUBYOPT: '-KU -E utf-8:utf-8'
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - name: Install Bundler
+        shell: bash
+        run: |
+          gem install bundler --version=$BUNDLE_VERSION --no-document
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            vendor/bundle
+            Gemfile.lock
+          key: process-balancer-bundle-${{ hashFiles('Gemfile.lock') }}
+          restore-keys: |
+            process-balancer-bundle-${{ hashFiles('Gemfile.lock') }}
+            process-balancer-bundle-
+
+      - name: 'Bundler install'
+        shell: bash
+        run: |
+          bundle _${BUNDLE_VERSION}_ check || bundle _${BUNDLE_VERSION}_ install --retry=$BUNDLE_RETRY
+
+      - uses: actions/cache@v2
+        with:
+          path: $HOME/.cache/rubocop_cache
+          key: process-balancer-rubocop-${{ hashFiles('.rubocop.yml') }}
+          restore-keys: |
+            process-balancer-rubocop-${{ hashFiles('.rubocop.yml') }}
+            process-balancer-rubocop-
+
+      - name: Rubocop
```

Comment:
Should we extract rubocop into a separate parallel step as we did in all other pipelines?

Reply:
Of course we can do it but in this case I do not see much benefits. Rubocop step takes ~4[s] please take a look here: https://github.com/NetsoftHoldings/process_balancer/runs/4285697235?check_suite_focus=true